contracts/liquidityMining/KyberFairLaunch.sol

```diff
@@ -0,0 +1,469 @@
+
+    //
+    // We do some fancy math here. Basically, any point in time, the amount of reward token
+    // entitled to a user but is pending to be distributed is:
+    //
+    //   pending reward = (user.amount * pool.accRewardPerShare) - user.rewardDebt
+    //
+    // Whenever a user deposits or withdraws Staking tokens to a pool. Here's what happens:
+    //   1. The pool's `accRewardPerShare` (and `lastRewardBlock`) gets updated.
+    //   2. User receives the pending reward sent to his/her address.
+    //   3. User's `amount` gets updated.
+    //   4. User's `rewardDebt` gets updated.
+  }
+
+  // Info of each pool
+  // stakeToken: token to stake, should be an ERC20 token
+  // startBlock: the block that the reward starts
+  // endBlock: the block that the reward ends
+  // rewardLockBps: lock percent (in bps) of reward that will be locked in the locker
+  // lastRewardBlock: last block number that rewards distribution occurs
+  // rewardPerBlock: amount of reward token per block
+  // accRewardPerShare: accumulated reward per share of token
+  struct PoolInfo {
+    address stakeToken;
+    uint32  startBlock;
+    uint32  endBlock;
+    uint32  rewardLockBps;
+    uint32  lastRewardBlock;
+    uint128 rewardPerBlock;
+    uint128 accRewardPerShare;
+  }
+
+  // contract for locking reward
+  IKyberRewardLocker public immutable rewardLocker;
+  // reward token
+  IERC20Ext public immutable rewardToken;
+
+  // Info of each pool.
+  PoolInfo[] public poolInfo;
+  // Info of each user that stakes Staking tokens.
+  mapping(uint256 => mapping(address => UserInfo)) public userInfo;
+
+  event AddNewPool(
+    address indexed stakeToken,
+    uint32 indexed startBlock,
+    uint32 indexed endBlock,
+    uint32 rewardLockBps,
+    uint256 rewardPerBlock
+  );
+  event RenewPool(
+    uint256 indexed pid,
+    uint32 indexed startBlock,
+    uint32 indexed endBlock,
+    uint32 rewardLockBps,
+    uint256 rewardPerBlock
+  );
+  event UpdatePool(
+    uint256 indexed pid,
+    uint32 indexed endBlock,
+    uint32 rewardLockBps,
+    uint256 rewardPerBlock
+  );
+  event Deposit(
+    address indexed user,
+    uint256 indexed pid,
+    uint256 indexed blockNumber,
+    uint256 amount
+  );
+  event Withdraw(
+    address indexed user,
+    uint256 indexed pid,
+    uint256 indexed blockNumber,
+    uint256 amount
+  );
+  event Harvest(
+    address indexed user,
+    uint256 indexed pid,
+    uint256 indexed blockNumber,
+    uint256 amount,
+    uint256 lockedAmount
+  );
+  event EmergencyWithdraw(
+    address indexed user,
+    uint256 indexed pid,
+    uint256 indexed blockNumber,
+    uint256 amount
+  );
+
+  constructor(
+    address _admin,
+    IERC20Ext _rewardToken,
+    IKyberRewardLocker _rewardLocker
+  ) PermissionAdmin(_admin) {
+    rewardToken = _rewardToken;
+    rewardLocker = _rewardLocker;
+
+    _rewardToken.safeApprove(address(_rewardLocker), uint256(-1));
```

Comment:
use `type(uint256).max` instead of `uint256(-1)`

Reply:
Updated